htmd/qm/psi4.py

```diff
@@ -0,0 +1,244 @@
+    >>> molFile = os.path.join(home('test-qm'), 'H2-0.74.mol2')
+    >>> mol = FFMolecule(molFile, method=FFTypeMethod.NONE) # doctest: +ELLIPSIS
+    <BLANKLINE>
+    ...
+
+    Create a Psi4 object
+    >>> from htmd.qm import Psi4
+    >>> qm = Psi4()
+    >>> qm # doctest: +ELLIPSIS
+    <htmd.qm.psi4.Psi4 object at 0x...>
+
+    Run single-point QM calculation of H2 with BLYP and cc-pVDZ
+    >>> from tempfile import TemporaryDirectory
+    >>> with TemporaryDirectory() as tmp:
+    ...     qm.molecule = mol
+    ...     qm.theory = 'BLYP'
+    ...     qm.basis = 'cc-pVDZ'
+    ...     qm.directory = tmp
+    ...     result = qm.run()
+
+    The QM results are returned as a list of htmd.qm.QMResult objects. See htmd.qm.QMResult documentation for details.
+    >>> result # doctest: +ELLIPSIS
+    [<htmd.qm.base.QMResult object at 0x...>]
+    >>> result[0].errored
+    False
+    >>> result[0].energy # doctest: +ELLIPSIS
+    -728.97082622...
+    >>> result[0].mulliken
+    [0.0, -0.0]
+
+    Run the geometry optimization of H2 with BLYP, but change basis to 3-21G.
+    NOTE: the `directory` attribut needs to be set to empty or non-existing directory, overwise the previous
+    calculation results are read.
+    >>> with TemporaryDirectory() as tmp:
+    ...     qm.basis = '3-21G'
+    ...     qm.optimize = True
+    ...     qm.directory = tmp
+    ...     result = qm.run()
+
+    The initial coordinates and optimize coordinates of H2 can be compared
+    >>> mol.coords
+    array([[[ 0.  ],
+            [ 0.  ],
+            [-0.37]],
+    <BLANKLINE>
+           [[ 0.  ],
+            [ 0.  ],
+            [ 0.37]]], dtype=float32)
+    >>> result[0].coords
+    array([[[ 0.        ],
+            [ 0.        ],
+            [-0.37527978]],
+    <BLANKLINE>
+           [[ 0.        ],
+            [ 0.        ],
+            [ 0.37527978]]])
+
+    The QM calculations run using LocalCPUQueue by default, but this can be changed to the others.
+    >>> from htmd.queues.acecloudqueue import AceCloudQueue
+    >>> qm.queue = AceCloudQueue() # doctest: +SKIP
+    """
+
+    @property
+    def _command(self):
+        return 'export HTMD_PSI4_WORKDIR=$(pwd)\npsi4 -i psi4.in -o psi4.out &> psi4.log'
+
+    def _completed(self, directory):
+        return os.path.exists(os.path.join(directory, 'psi4.out'))
+
+    def _writeInput(self, directory, iframe):
+
+        with open(os.path.join(directory, 'psi4.in'), 'w') as f:
+
+            f.write('set_num_threads( %d )\n' % self._ncpus)
+            f.write('memory %f gb\n\n' % self._mem)
+
+            reference = 'r' if self.multiplicity == 1 else 'u'
+            reference += 'hf' if self.theory == 'RHF' else 'ks'
+            f.write('set { reference %s }\n' % reference)
+            f.write('set { basis %s }\n\n' % self.basis)
+
+            if self.solvent == 'vacuum':
+                pass
+            elif self.solvent == 'PCM':
+                # TODO check all agruments
```

Comment:
Is it crucial for now or can be left for later?

Reply:
Later!